### ISSUE TITLE
`algorithm.sort` should have a default value for `cmp`

### ISSUE THREAD

**Issue by tim-st** [nim-lang/Nim#8684]
Currently `algorithm.sort` doesn't have a default value for the compare proc which needs to get passed.
Often `system.cmp` is wanted as cmp proc by the user and it's a bit annoying to always call
`a.sort(cmp)` or `a.sort(cmp=cmp)` instead of just `a.sort()` if that would work using `system.cmp` and no explicit cmp proc is passed.

Python has the default signature, too:
```Python
>>> a = [6,5,4]
>>> a.sort()
>>> a
[4, 5, 6]
```

The current signature is
```nim
proc sort*[T](a: var openArray[T], cmp: proc (x, y: T): int {.closure.}, order = SortOrder.Ascending) =
```

The same should apply for `sorted` and `isSorted`.

**Comment by GULPF** [comment]
I think instead of a default value for the parameter, there should be an overload of `sort`  without any `cmp` argument at all. That overload can simply call `cmp` directly, which will be resolved to whatever implementation of `cmp` is available in the scope (typically `system.cmp`).

**Comment by tim-st** [comment]
I agree it could be implemented using overloading. Though it seems quite rare to me that a proc with exactly the same name and signature is in the scope other than `system.cmp`.
I think it's only easy to implement, if also [this](https://github.com/nim-lang/Nim/issues/8683) is considered easy or the bug mentioned is not affected by overloaded procs and the solution here is overloaded.

**Comment by citycide** [comment]
Is this closed by #9174?

**Comment by Araq** [comment]
Yup.
